postgres/src/main/java/org/jdbi/v3/postgres/TypedEnumArgumentFactory.java

```diff
@@ -13,25 +13,27 @@
  */
 package org.jdbi.v3.postgres;
 
+import java.lang.reflect.Type;
 import java.sql.Types;
+import java.util.Optional;
 
-import org.jdbi.v3.core.argument.AbstractArgumentFactory;
 import org.jdbi.v3.core.argument.Argument;
+import org.jdbi.v3.core.argument.ArgumentFactory;
 import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.generic.GenericTypes;
 
 /**
  * Default {@code jdbi} behavior is to bind {@code Enum} subclasses as
  * a string, which Postgres won't implicitly convert to an enum type.
  * If instead you bind it as {@code java.sql.Types.OTHER}, Postgres will
  * autodetect the enum correctly.
  */
-public class TypedEnumArgumentFactory extends AbstractArgumentFactory<Enum<?>> {
```

Comment:
Wouldn't it better to fix the issue just by switching `Enum<?>` to `Enum`? This way we won't compare `ParametrizedType` instances, but will check that an enum can be assigned to the `Enum` class.